File: Punto4/suma.py
import time

start_time1 = time.time()
suma1 = 0
for i in range(1,99999):
	suma1 += i
end_time1 = time.time()

start_time2 = time.time()
suma2 = 0
for i in range(1,999999):
        suma2 += i
end_time2 = time.time()

start_time3 = time.time()
suma3 = 0
for i in range(1,9999999):
        suma3 += i
end_time3 = time.time()

print(f"Tiempo de ejecución en Python de la suma de 1 a 99999: {end_time1 - start_time1:.6f} segundos")
print(f"Tiempo de ejecución en Python de la suma de 1 a 999999: {end_time2 - start_time2:.6f} segundos")
print(f"Tiempo de ejecución en Python de la suma de 1 a 9999999: {end_time3 - start_time3:.6f} segundos")

File: Punto4/suma.c
#include <stdio.h>
#include <time.h>

int main() {
    clock_t start1, end1, start2, end2, start3, end3;
    double time1, time2, time3;
    
    start1 = clock();
    
    long long sum1 = 0;
    for (int i = 1; i <= 99999; i++) {
        sum1 += i;
    }
    
    end1 = clock();
    time1 = ((double) (end1 - start1)) / CLOCKS_PER_SEC;
    
    printf("Tiempo de ejecución en C de la suma de 1 a 99999: %f segundos\n", time1);

    start2 = clock();
    
    long long sum2 = 0;
    for (int i = 1; i <= 999999; i++) {
        sum2 += i;
    }
    
    end2 = clock();
    time2 = ((double) (end2 - start2)) / CLOCKS_PER_SEC;
    
    printf("Tiempo de ejecución en C de la suma de 1 a 999999: %f segundos\n", time2);

    start3 = clock();
    
    long long sum3 = 0;
    for (int i = 1; i <= 9999999; i++) {
        sum3 += i;
    }
    
    end3 = clock();
    time3 = ((double) (end3 - start3)) / CLOCKS_PER_SEC;
    
    printf("Tiempo de ejecución en C de la suma de 1 a 9999999: %f segundos\n", time3);


    return 0;
}
